### ISSUE TITLE
Discuss page title rule

### ISSUE THREAD

**Issue by AndersEkl** [nlbdev/nordic-accessible-epub-guidelines#6]
The KB recommends having detailed page titles, including chapter etc. In our guidelines we require identical page titles for each content document. Is this something we should reconsider?

**Comment by AndersEkl** [comment]
https://www.w3.org/TR/epub-a11y-tech-11/#titles-001

**Comment by AndersEkl** [comment]
I am thinking this rule should be something like `<title>Chapter/Part number, title (if there is one) - Book title</title>`. And obviously, content documents containing other parts of the book, like the colophon, will have "Colophon" instead of chapter.

**Comment by AndersEkl** [comment]
@martinpub @jonaslil @oscarlcarlsson Input on this?

**Comment by AndersEkl** [comment]
I agree in that it probably has a very limited, if any, practical meaning in an EPUB context for our users. But on the other hand, it is part of the EPUB Accessibility Techniques 1.1 specification.

**Comment by josteinaj** [comment]
> On the other hand, @josteinaj mentioned allowing their patrons to download books as html. Adding more descriptive titles might make sense in that specific use case.

The HTML we distribute is a single HTML file. We concatenate the contents of all the `<body>` elements in spine order. So for our use we can have all the `<title>` elements be the same.

Generating this metadata may have to be done in various tools as well (narration, format conversion, etc.), if you want to get a valid nordic EPUB as output. So if the `<title>` elements should be more detailed, I think it should be as easy as possible to generate them. For instance just use the text from the first navigable headline in the document. If there's no navigable headline, then use the book title. In most cases, this will result in using the chapter headline.

**Comment by martinpub** [comment]
> * In my (admittedly limited) experience, screen readers ignore the title element of content documents in an epub. This is probably a good thing: you would not want to hear the main heading twice.

Are you sure? It's both in the DAISY KB (I don't read the context as non-EPUB, but rather, HTML in EPUB) and EPUB Accessibility Techniques.

I interpret this as current best practice.

**Comment by AndersEkl** [comment]
From MTMs guidelines:
For bodymatter sections where no heading is available in the source copy, the 
Nordic Guidelines require labels to be created and stored in an @aria-label
attribute in the heading section. For MTM productions, if nothing else is 
mentioned in accompanying Editing Instructions, we kindly request our 
producers to name the untitled sections using the first three words of the 
chapter text, followed by a space and an ellipsis, e.g. aria-label="Det var 
längesedan …" for a bodymatter section where the first three words of the 
body text are "Det var längesedan".

**Comment by jonaslil** [comment]
We agreed in the editor's meeting today to include a requirement for a more descriptive title based on the main heading or (if no heading is present) the aria-label of the main section.

As to the question by @martinpub, I have tested epubs in Thorium + NVDA, ADE + NVDA and Apple Books + Voice Over. None of them reads the page title. There are of course many more combinations you could try.

**Comment by martinpub** [comment]
> As to the question by @martinpub, I have tested epubs in Thorium + NVDA, ADE + NVDA and Apple Books + Voice Over. None of them reads the page title. There are of course many more combinations you could try.

Did you try focusing on another application/window, then returning to the reading app?

I tried NVDA + Thorium, and sometimes I get the title element announced, but not sure why I cannot reproduce it all the time.
